src/sentry/monitors/endpoints/organization_monitor_stats.py

```diff
@@ -60,28 +53,57 @@ def get(self, request: Request, organization, project, monitor) -> Response:
         environments = get_environments(request, organization)
 
         if environments:
-            history = history.filter(monitor_environment__environment__in=environments)
-
-        for dt, status, duration in history.values_list(
-            "date_added", "status", "duration"
-        ).iterator():
-            ts = normalize_to_epoch(dt, args["rollup"])
-            stats[ts][status] += 1
-            if duration:
-                duration_stats[ts]["sum"] += duration
-                duration_stats[ts]["num_checkins"] += 1
-
-        stats_duration_data = []
-        statuses_to_name = dict(CheckInStatus.as_choices())
-        # compute average duration and construct response object
-        for ts, data in stats.items():
-            duration_sum, num_checkins = duration_stats[ts].values()
-            avg_duration = 0
-            if num_checkins > 0:
-                avg_duration = duration_sum / num_checkins
-            datapoint = {statuses_to_name[status]: data[status] for status in tracked_statuses}
-            datapoint["ts"] = ts
-            datapoint["duration"] = avg_duration
-            stats_duration_data.append(datapoint)
-
-        return Response(stats_duration_data)
+            check_ins = check_ins.filter(monitor_environment__environment__in=environments)
+
+        # Use postgres' `date_bin` to bucket rounded to our rolllups
+        bucket = Func(
+            timedelta(seconds=args["rollup"]),
+            "date_added",
+            datetime.fromtimestamp(end),
+            function="date_bin",
```

Comment:
This is only available in postgres 14. I think ST is still on an older version of postgres, meaning we'll have to wait to turn on crons in ST until it is.

Reply:
Yes, ST is mostly on older postgres.